src/Microsoft.ML.TorchSharp/Extensions/IEnumerableExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.ML.TorchSharp.Extensions
+{
+    internal static class IEnumerableExtensions
+    {
+        public static string ToString<T>(this IEnumerable<T> enumerable, bool verbose = true)
+        {
+            return verbose
+                ? $"[{string.Join(", ", enumerable.Select(e => e.ToString()))}]"
+                : enumerable?.ToString() ?? string.Empty;
+        }
+
+        public static T ArgMin<T>(this IEnumerable<T> source, Func<T, int> getValue)
```

Comment:
>ArgMin<T>(this IEnumerable<T> source, Func<T, int> getValue) [](http://example.com/codeflow?start=24&length=60)

can't this method implemented without any allocations? just not use Linq and do simple loop 


```C#

foreach (T t in source)
{
   int value = getValue(t);
   if (value < minValue)
   {
       minValue = value; 
       minSource = t;
   }

....
}
```

Reply:
I'm not entirely sure honestly. This is used by the BPE tokenizer and I didn't edit that file much. When we get your new tokenizer this can go away entirely.

Reply:
So I'm gonna leave it as is for now rather than risk changing it as it will go away.